chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -695,13 +952,16 @@ where
             &self.utxo_cache,
             tx,
             SignatureDestinationGetter::new_for_transaction(
-                &self.accounting_delta_adapter.accounting_delta(),
+                &self.tokens_accounting_cache,
```

Comment:
Please add tests that use an invalid signature, to ensure that destination verification is being done correctly.